### PR TITLE
[4.3] Codemirror corrects path for keybinding script +(obey debug - versioning)

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue #40049 .

### Testing Instructions

- make codemirror the default browser
- Enable keymaping
- Check that codemirro and keybinding work

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: plugins/editors/codemirror/codemirror.php [hunk 1]
```diff
@@ -14,6 +14,7 @@
 use Joomla\CMS\Plugin\CMSPlugin;
 use Joomla\CMS\Plugin\PluginHelper;
 use Joomla\Event\Event;
+use Joomla\CMS\Uri\Uri;
```
Comment (Quy) [inline plugins/editors/codemirror/codemirror.php]:
```suggestion
use Joomla\CMS\Uri\Uri;
use Joomla\Event\Event;
```

**Comment by brianteeman** [comment]
This only partially works. The js needs to be updated as well

```
    if (this.options.keyMapUrl) {
      await import(`${this.options.keyMapUrl}`);
    }
```

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on edf19de76e388cf0906b0ab2e85db3d42d6adb7c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40051">issues.joomla.org/tracker/joomla-cms/40051</a>.</sub>

**Comment by dgrammatiko** [comment]
@brianteeman I just realised that the script is not versioned (it cannot be invalidated if there's a new version) and it's easier to patch this here than opening a new PR. If that's not ok I will revert the last commit

**Comment by brianteeman** [comment]
I always prefer one pr per task but its your pr so its your choice. I just ask that if you keep the media version stuff that you update the title and the test instructions please.

**Comment by Quy** [comment]
> Call to a member function getDocument() on null

File: plugins/editors/codemirror/codemirror.php [hunk 2]
```diff
@@ -244,7 +245,8 @@ public function onDisplay(
         }
 
         if ($options->keyMap !== 'default') {
-            $keyMapUrl = $this->basePath . 'keymap/' . $options->keyMap . '.min.js';
+            $keyMapUrl = HTMLHelper::_('script', $this->basePath . 'keymap/' . $options->keyMap . '.min.js', ['relative' => false, 'pathOnly' => true]);
+            $keyMapUrl .= '?' . $this->getApplication()->getDocument()->getMediaVersion();
```
Comment (Quy) [inline plugins/editors/codemirror/codemirror.php]:
```suggestion
            $keyMapUrl .= '?' . $this->app->getDocument()->getMediaVersion();
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on d1117f48ab313747c145eb35e31aa87a7017c077<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40051">issues.joomla.org/tracker/joomla-cms/40051</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on d1117f48ab313747c145eb35e31aa87a7017c077<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40051">issues.joomla.org/tracker/joomla-cms/40051</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40051">issues.joomla.org/tracker/joomla-cms/40051</a>.</sub>

**Comment by HLeithner** [comment]
This pull request has been automatically rebased to 4.3-dev.

**Comment by obuisard** [comment]
Hello Dimitris @dgrammatiko. Do you mind checking the conflict?

**Comment by dgrammatiko** [comment]
@obuisard done

**Comment by obuisard** [comment]
> @obuisard done

Thanks Dimitris @dgrammatiko , appreciated!

**Comment by obuisard** [comment]
@Quy or Vio @viocassel, could any of you give it another test as this PR has been re-based from 4.2 to 4.3 and the code moved around under the new structure for plugins. Thank you!

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 04ba454907b0bf228464be2973d2c78a3317fa4c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40051">issues.joomla.org/tracker/joomla-cms/40051</a>.</sub>

**Pull request merged**